tools/unicode_properties_parse/grapheme_break_property_data_gen.py

```diff
@@ -291,28 +302,58 @@ def generate_data_tables() -> str:
     DerivedCoreProperties.txt can be found at
     https://www.unicode.org/Public/UCD/latest/ucd/DerivedCoreProperties.txt
 
+    EastAsianWidth.txt can be found at
+    https://www.unicode.org/Public/UCD/latest/ucd/EastAsianWidth.txt
+
     All files are expected to be in the same directory as this script.
     """
     gbp_filename, gbp_timestamp, gbp_ranges = read_file("GraphemeBreakProperty.txt")
     emoji_filename, emoji_timestamp, emoji_ranges = read_file("emoji-data.txt")
     cat_filename, cat_timestamp, cat_ranges = read_file("DerivedGeneralCategory.txt")
     derived_filename, derived_timestamp, derived_ranges = read_file("DerivedCoreProperties.txt")
+    eaw_filename, eaw_timestamp, eaw_ranges = read_file("EastAsianWidth.txt")
 
     printable_ranges = compact_property_ranges(sorted([
         PropertyRange(x.lower, x.upper, "Yes")
         for x in cat_ranges
         if x.prop not in ('Cc', 'Cf', 'Cs', 'Co', 'Cn', 'Zl', 'Zp', 'Zs') or chr(x.lower) == ' '
     ], key=lambda x: x.lower))
 
+    std_wide_ranges = [
+        range(0x4DC0, 0x4DFF),
+        range(0x1F300, 0x1F5FF),
+        range(0x1F900, 0x1F9FF),
+    ]
+
+    def has_width_2(prop_range):
+        if prop_range.prop in ("F", "W"):

```

Comment:
Question: Both `EastAsianWidth-15.0.0.txt` and `EastAsianWidth-15.1.0.txt` said:

```
#  - The unassigned code points in the following blocks default to "W":
#         CJK Unified Ideographs Extension A: U+3400..U+4DBF
#         CJK Unified Ideographs:             U+4E00..U+9FFF
#         CJK Compatibility Ideographs:       U+F900..U+FAFF
#  - All undesignated code points in Planes 2 and 3, whether inside or
#      outside of allocated blocks, default to "W":
#         Plane 2:                            U+20000..U+2FFFD
#         Plane 3:                            U+30000..U+3FFFD
```

We previously had code corresponding to this:

https://github.com/microsoft/STL/blob/8b081e26ba016970ce3338cb483ff10f2ade30a5/tools/unicode_properties_parse/format_width_estimate_intervals.py#L112-L119

I don't see that in the new implementation. Will that be a problem in the future? I observe that this doesn't affect the output for the 15.0.0 file, though.

Reply:
I found that the data file already handles this, by specifying every code point in these ranges as Wide.

For example, these are the relevant lines in `EastAsianWidth-15.0.0.txt`:
```
3400..4DBF;W     # Lo  [6592] CJK UNIFIED IDEOGRAPH-3400..CJK UNIFIED IDEOGRAPH-4DBF
4E00..9FFF;W     # Lo [20992] CJK UNIFIED IDEOGRAPH-4E00..CJK UNIFIED IDEOGRAPH-9FFF
F900..FA6D;W     # Lo   [366] CJK COMPATIBILITY IDEOGRAPH-F900..CJK COMPATIBILITY IDEOGRAPH-FA6D
FA6E..FA6F;W     # Cn     [2] <reserved-FA6E>..<reserved-FA6F>
FA70..FAD9;W     # Lo   [106] CJK COMPATIBILITY IDEOGRAPH-FA70..CJK COMPATIBILITY IDEOGRAPH-FAD9
FADA..FAFF;W     # Cn    [38] <reserved-FADA>..<reserved-FAFF>
20000..2A6DF;W   # Lo [42720] CJK UNIFIED IDEOGRAPH-20000..CJK UNIFIED IDEOGRAPH-2A6DF
2A6E0..2A6FF;W   # Cn    [32] <reserved-2A6E0>..<reserved-2A6FF>
2A700..2B739;W   # Lo  [4154] CJK UNIFIED IDEOGRAPH-2A700..CJK UNIFIED IDEOGRAPH-2B739
2B73A..2B73F;W   # Cn     [6] <reserved-2B73A>..<reserved-2B73F>
2B740..2B81D;W   # Lo   [222] CJK UNIFIED IDEOGRAPH-2B740..CJK UNIFIED IDEOGRAPH-2B81D
2B81E..2B81F;W   # Cn     [2] <reserved-2B81E>..<reserved-2B81F>
2B820..2CEA1;W   # Lo  [5762] CJK UNIFIED IDEOGRAPH-2B820..CJK UNIFIED IDEOGRAPH-2CEA1
2CEA2..2CEAF;W   # Cn    [14] <reserved-2CEA2>..<reserved-2CEAF>
2CEB0..2EBE0;W   # Lo  [7473] CJK UNIFIED IDEOGRAPH-2CEB0..CJK UNIFIED IDEOGRAPH-2EBE0
2EBE1..2F7FF;W   # Cn  [3103] <reserved-2EBE1>..<reserved-2F7FF>
2F800..2FA1D;W   # Lo   [542] CJK COMPATIBILITY IDEOGRAPH-2F800..CJK COMPATIBILITY IDEOGRAPH-2FA1D
2FA1E..2FA1F;W   # Cn     [2] <reserved-2FA1E>..<reserved-2FA1F>
2FA20..2FFFD;W   # Cn  [1502] <reserved-2FA20>..<reserved-2FFFD>
30000..3134A;W   # Lo  [4939] CJK UNIFIED IDEOGRAPH-30000..CJK UNIFIED IDEOGRAPH-3134A
3134B..3134F;W   # Cn     [5] <reserved-3134B>..<reserved-3134F>
31350..323AF;W   # Lo  [4192] CJK UNIFIED IDEOGRAPH-31350..CJK UNIFIED IDEOGRAPH-323AF
323B0..3FFFD;W   # Cn [56398] <reserved-323B0>..<reserved-3FFFD>
```

Reply:
Excellent, thanks!